RELEASE_NOTES.rst

```diff
@@ -21,7 +21,204 @@
 
 .. towncrier release notes start
 
-Airflow 2.7.3 (2023-11-04)
+Airflow 2.8.0 (2023-12-14)
```

Comment:
Should this be December 18th instead?

Reply:
I'll update it in a separate PR so we can cherrypick it to 2.8 branch and build docs